### PR TITLE
Support safe area views

### PR DESCRIPTION
**Pull request by adamstep**
Addresses Issue #212 .

- Adds a `safe-area` attribute to view-like elements, most notably `<view>`, `<body>`, and `<header>`.
  - **NOTE**: `safe-area` will not have effect if the view also has `scroll="true"`.
- Updated the example app to use `safe-area` instead of padding through styling.
- Updated docs to explain the new attribute.
- Updated XML schema to accept the new attribute.

### Screenshots

The same XML with `safe-area="true"`, as viewed on iPhone 8 and 11 (with notch)
Screen Shot 2021-01-20 at 9 03 30 AM
Screen Shot 2021-01-20 at 9 13 35 AM

### CODE REVIEW

File: examples/case_studies/basic_form/index.xml [hunk 1]
```diff
@@ -56,7 +56,7 @@
              marginTop="16" />
     </styles>
     <body scroll="true"
-          style="Body">
+          style="Body" >
```
Comment (flochtililoch) [inline examples/case_studies/basic_form/index.xml:59]:
```suggestion
          style="Body" safe-area="true">
```

File: examples/case_studies/lazy_load/index.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
       <style id="Body"
              backgroundColor="white"
              flex="1"
-             paddingTop="48" />
+              />
```
Comment (flochtililoch) [inline examples/case_studies/lazy_load/index.xml:7]: I think the `<body>` tag lower down this file should also set `safe-area` attribute

Comment (adamstep) [reply examples/case_studies/lazy_load/index.xml:7]: The design of this screen is different from the others, it feature big colorful blocks that should not use `safe-area`.

**Comment by adamstep** [comment]
@flochtililoch ready to re-review, Flow is passing now

**Review by flochtililoch — APPROVED** [review]

**Comment by cengit** [comment]
awesome

**Pull request merged**